ddtrace/tracer/option.go

```diff
@@ -302,8 +305,19 @@ const partialFlushMinSpansDefault = 1000
 func newConfig(opts ...StartOption) *config {
 	c := new(config)
 	c.sampler = NewAllSampler()
+	defaultRate, err := internal.FloatVal(getDDorOtelConfig("sampleRate"), math.NaN())
+	if err != nil {
+		log.Warn("ignoring DD_TRACE_SAMPLE_RATE, error: %v", err)
+	} else if defaultRate < 0.0 || defaultRate > 1.0 {
+		log.Warn("ignoring DD_TRACE_SAMPLE_RATE: out of range %f", defaultRate)
+		defaultRate = math.NaN()
+	}
+	c.globalSampleRate = defaultRate
```

Comment:
Not blocking nit: I feel this could have been kept inside the globalSampleRate() function. In any case, if next quarter I can push for my idea expressed [here](https://github.com/DataDog/dd-trace-go/pull/2715/files#r1622426041), it'll be refactored 😄

Reply:
I thought storing the globalSampleRate on the config object would be easier to maintain and provide better performance. `globalSampleRate()` would parse the environment variables on every invocation.  I don't have strong feeling here. It's an easy change to revert.

Reply:
@mabdinur No need to revert, it's fine.